### ISSUE TITLE
External definitions

### ISSUE THREAD

**Issue by gregbacchus** [touchifyapp/spec2ts#1]
I have one main JSON Schema file which references definitions in a second, and the second file has only definitions.

e.g.

defs.schema.json
```json
{
  "$id": "http://example.com/schemas/defs.schema.json",
  "definitions": {
    "int": {
      "type": "integer"
    },
    "str": {
      "type": "string"
    }
  }
}
```

and
test.schema.json
```json
{
  "$id": "http://example.com/schemas/test.schema.json",
  "type": "object",
  "title": "MyObj",
  "properties": {
    "foo": {
      "$ref": "defs.schema.json#/definitions/int"
    },
    "bar": {
      "$ref": "defs.schema.json#/definitions/str"
    }
  }
}
```

There are a couple of issues when converting these files.

1. The file that only contains the definitions: none of the interfaces are generated, i.e. it's only

```typescript
export type Defs = any;
```

2. The dependent file doesn't reference the other correctly, i.e.

```typescript
import { Int } from "defs.schema";
import { Str } from "defs.schema";
```

should be

```typescript
import { Int } from "./defs.schema.d";
import { Str } from "./defs.schema.d";
```

or better

```typescript
import { Int, Str } from "./defs.schema.d";
```

**Comment by SomaticIT** [comment]
Hello,

Thank you for trying this library and reporting this issue.

We will try to reproduce the bugs. If I'm correct, here are the issues when need to cover:
- Enforce relative references to start with `./`
- Try to merge imports when multiple definitions are used from the same reference.
- Always generate types from `definitions`

**Comment by SomaticIT** [comment]
@gregbacchus These issues were fixed in release `@spec2ts/jsonschema@1.3.1`.
Can you try it and confirm that it works for you?

Thanks
